launcher/src/main/java/org/apache/spark/launcher/InProcessAppHandle.java

```diff
@@ -63,6 +71,7 @@ synchronized void start(String appName, Method main, String[] args) {
         main.invoke(null, (Object) args);
       } catch (Throwable t) {
         LOG.log(Level.WARNING, "Application failed with exception.", t);
+        error = t;
```

Comment:
Might be good to special-case `InvocationTargetException` and avoid one level of wrapping which is probably not useful to anyone.